content/md/en/docs/test/unit-testing.md

```diff
@@ -113,6 +113,56 @@ Some things to note are:
 - Events are not emitted on the genesis block, and so the block number should be set in order for this test to pass.
 - You need to have a `.into()` after instantiating your pallet event, which turns it into a generic event.
 
+### Advanced Event Testing
+
+When testing events in a pallet, often you are only interested in the events that are emitted from your own pallet.
+A helper function like the one below will filter the events to only the ones emitted by your chain, and also convert them into your own event type. This is usually placed in the mock file, `mock.rs`.
+
+```rust
+fn only_example_events() -> Vec<super::Event<Runtime>> {
+ System::events()
+  .into_iter()
+  .map(|r| r.event)
+  .filter_map(|e| if let RuntimeEvent::TemplateModule(inner) = e { Some(inner) } else { None })
+  .collect::<Vec<_>>();
+}
+```
+
+Additionally, if your test performs multiple operations that emit events in sequence, it may be preferable to only see the events that have happened since the last check. The example below leverages the helper function above.
```

Comment:
```suggestion
Additionally, if your test performs operations that emit events in a sequence, you might want to only see the events that have happened since the last check. 
The following example leverages the preceding helper function.
```